### PR TITLE
WSTEAMA-341 Metadata Component for new Home Page

### PR DESCRIPTION
**Pull request by LilyL0u**
Resolves https://jira.dev.bbc.co.uk/browse/WSTEAMA-341

Overall changes
======
Adds the MetaData component to our new HomePage, passing the values previously outlined in [WSTEAMA-321](https://jira.dev.bbc.co.uk/browse/WSTEAMA-321)

Code changes
======

- MetaData component added to `src/app/pages/HomePage/HomePage.tsx`
- Some small refactor work to destructure existing data.
- Adds a unit test to check for the metadata title.
- Adds a unit test to check for the metadata description.

Testing
======

Helpful Links
======

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/components/Metadata/types.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export interface MetadataProps {
   socialHeadline?: string;
   lang: string;
   twitterHandle?: string;
-  description: string;
+  description: string | undefined;
```
Comment (karinathomasbbc) [inline src/app/components/Metadata/types.ts]: I think we should now be able to revert this change

**Review by karinathomasbbc — APPROVED** [review]
Awesome @louisearchibald, thanks for picking up where @LilyL0u left off. 

It'll be great to see metadata on our homepage, and this also makes testing it on mobile so much better!

**Review by amoore108 — APPROVED** [review]

File: src/app/components/Metadata/types.ts (hunk 1 above)
Comment (louisearchibald) [reply src/app/components/Metadata/types.ts]: This line complains when I revert it ->

https://github.com/bbc/simorgh/blob/91ef395cc80fa09675dc34c43f5cc7d3adfc4acf/src/app/pages/HomePage/HomePage.tsx#L41

Would you assert the type by having `description={description as string}` or is there a better way around it?

Comment (karinathomasbbc) [reply src/app/components/Metadata/types.ts]: If we were to make it optional, would that work? But then we could end up in a situation where description is `undefined` on the homepage, and that's not great...

Comment (karinathomasbbc) [reply src/app/components/Metadata/types.ts]: Do you know what is returned from the bff when the description in Tipo is empty? Do we get back an empty string instead?

Comment (louisearchibald) [reply src/app/components/Metadata/types.ts]: I believe so. I'll clear out the field in Tipo and see what comes back to double check.

Comment (louisearchibald) [reply src/app/components/Metadata/types.ts]: Empty string!

Comment (karinathomasbbc) [reply src/app/components/Metadata/types.ts]: So I guess this means that we should always have a description, and maybe it's worth defaulting the value to `''` as we're destructuring it from the pageData object?

File: src/app/pages/HomePage/HomePage.tsx [hunk 2]
```diff
@@ -11,24 +11,34 @@ import {
...
     title: string;
     curations: CurationData[];
+    description?: string;
   };
 }
 
 const HomePage = ({ pageData }: HomePageProps) => {
-  const { curations } = pageData;
-  const { translations, product, serviceLocalizedName } =
+  const { curations, description } = pageData;
```
Comment (karinathomasbbc) [inline src/app/pages/HomePage/HomePage.tsx:27]: Let's do this, so we can remove the `description: string | undefined;` in src/app/components/Metadata/types.ts

```suggestion
  const { curations, description = '' } = pageData;
```

File: src/app/pages/HomePage/HomePage.tsx [hunk 3]
```diff
@@ -11,24 +11,34 @@ import {
...
 import styles from './index.styles';
+import MetadataContainer from '../../components/Metadata';
 
 interface HomePageProps {
   pageData: {
     pageType: string;
     id?: string;
     title: string;
     curations: CurationData[];
+    description?: string;
```
Comment (karinathomasbbc) [inline src/app/pages/HomePage/HomePage.tsx]: On second thoughts, we might just need to set this to be a mandatory instead of optional prop, and therefore don't need to default it when destructuring

Comment (louisearchibald) [reply src/app/pages/HomePage/HomePage.tsx]: Changed!
